lxd/warnings.go

```diff
@@ -362,7 +362,12 @@ func warningPut(d *Daemon, r *http.Request) response.Response {
 	}
 
 	err = s.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
-		err := tx.UpdateWarningStatus(id, status)
+		_, err := cluster.GetWarning(ctx, tx.Tx(), id)
+		if err != nil {
+			return err
+		}
+
+		err = tx.UpdateWarningStatus(id, status)
```

Comment:
@gabrielmougard please can you look into why this function call does not return a not found error when the `id` doesn't exist. We shouldn't need to use `GetWarning` first.